client/app/models/project.js

```diff
@@ -124,4 +125,19 @@ export default class ProjectModel extends Model {
     return this.artifacts.sortBy('modifiedon')
       .reverse();
   }
+
+  @sort('milestones', function(prev, next) {
+    const milestoneSequenceDifference = prev.dcpMilestonesequence - next.dcpMilestonesequence;
+
+    if (milestoneSequenceDifference === 0) {
+      if (!prev.displayDate)
+        return 1;
+      if (!next.displayDate)
+        return -1;
+      return prev.displayDate - next.displayDate;
+    }
+
+    return milestoneSequenceDifference;
+  })
+  sortedMilestones;
```

Comment:
Previously, the backend sorted immediately after calculating displayDate/displayDate2. Now that those properties are on the frontend, we also sort on the frontend. Sorting on the frontend make more sense, too, because it is related to data display.

This is essentially a migration of the [`sortMilestones` function from the backend to frontend](https://github.com/NYCPlanning/labs-zap-search/pull/1290/files#diff-3cb281dcf0b415503e45ea663c63a42115e63a34eea5df328a7a7dad7af349eaL234-L241)

Reply:
Great